### PR TITLE
ASR improvements

### PR DESCRIPTION
**Pull request by titu1994**
# Changelog
- Update Citrinet configs.
- Improved error message when numba is not installed.
- Add tarred dataset support for all supported filetypes.
- Ignore extra parameters for layer-normed LSTM.
- Correct deprecated call signature for OmegaConf.

### CODE REVIEW

File: examples/asr/conf/citrinet/citrinet_384.yaml [hunk 1]
```diff
@@ -11,30 +20,34 @@ model:
...
     shuffle: false
+    use_start_end_token: false
 
   model_defaults:
     repeat: 5
     dropout: 0.0
     separable: true
     se: true
     se_context_size: -1
+    kernel_size_factor: 1.0
```
Comment (VahidooX) [inline examples/asr/conf/citrinet/citrinet_384.yaml:50]: Do you use these model_defaults in the code directly or they are just some place to keep the defaults?

Comment (titu1994) [reply examples/asr/conf/citrinet/citrinet_384.yaml:50]: model defaults are used in config directly + used by code to know what is globally shared across the configs. Especially useful for RNNT configs. Here, setting the model_defaults.dropout would enable dropout on most of the blocks of Citrinet so its easier for the user.

Comment (VahidooX) [reply examples/asr/conf/citrinet/citrinet_384.yaml:50]: It is fine to have it. Just wanted to make sure that our code is reading these values and set the appropriate places.
If we drop all kernel_size_factor in the rest of the config, does it use this value.

Comment (titu1994) [reply examples/asr/conf/citrinet/citrinet_384.yaml:50]: Oh no - this value is passed by omegaconf resolution only. Later on this value can be changed but the rest of the config will not be updated. 

Hmm that's a good point, guess I should make it forced propagated via code. But it breaks one niche scenario. Hmm will fix that after some consideration in another pr

Comment (VahidooX) [reply examples/asr/conf/citrinet/citrinet_384.yaml:50]: The option of not propagating is aslo not a bad choice. We would have them as a place holder in the config to get propagated by omegaconf. In this case, no need to worry about conflicts on our side.

Comment (titu1994) [reply examples/asr/conf/citrinet/citrinet_384.yaml:50]: Yeah, best leave it to omegaconf to handle

**Pull request merged**